First review

ConstantineEpifanov opened the pull request.

ConstantineEpifanov committed:
chore: delete default stories and rewrite index

vladdevjs committed:
fix: many fixes in Footer / added events fetching

mintolime committed:
feat: added validation hooks and edited inputs props

vladdevjs committed:
feat: fill map with data

ConstantineEpifanov committed:
feat: add Like/Dislike on PointItem

sergasent committed:
feat(buttons-list): multiselect and column-template for ButtonsList

sergasent committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

sergasent committed:
feat(booking): made TabSwitcher

mintolime committed:
dev: added hooks popupOpen and redacted popup

mintolime committed:
dev: first attempt to set up auth-tion and regi-tion

vladdevjs committed:
fix: responsive Events slider and some fixes

ConstantineEpifanov committed:
fix: fix api on PointItem

mintolime committed:
fix:  fixed validation hooks

ConstantineEpifanov committed:
feat: add More button to Main page

ConstantineEpifanov committed:
feat: add api and routing to Coworking own page

ConstantineEpifanov committed:
fix: fix styles

mintolime committed:
fix: refactored validation hook and added validation of imput and buttons

vladdevjs committed:
feat: added Context

vladdevjs committed:
fix: fixed Map component

mintolime committed:
fix: changing the func. of reg-tion and auth-tion

ConstantineEpifanov committed:
feat: add More button to coworkingList page and Back button to Coworking/fix styles

mintolime committed:
fix: added validation for confirm code form

ConstantineEpifanov committed:
chore: add docker action

ConstantineEpifanov committed:
chore: fix workflow

sergasent committed:
feat: add plan photo and TabSwitcher for Booking

sergasent committed:
fix(tabswitcher): change TabSwither layout

vladdevjs committed:
feat: Favorites Tab logic and Map fixes

vladdevjs committed:
Merge branch 'develop' of github.com:ConstantineEpifanov/book-it-coworking-frontend into develop

sergasent committed:
feat(booking): make main layout

sergasent committed:
feat(booking): make extra rules checking for the Calendar

sergasent committed:
feat(booking): create extra rules

ConstantineEpifanov committed:
chore: fix dockerfile

mintolime committed:
fix: fixed token func and added info profile in the header

mintolime committed:
Merge branch 'develop' into auth-functionality

mintolime committed:
test: test commit

sergasent committed:
fix: change active button cursor for TabSwitcher

mintolime committed:
refactor: corrected popup routers and edited styles

sergasent committed:
feat(booking): made main logic of Booking page

ConstantineEpifanov committed:
fix: fix more reviews button, rules render and rating color

ConstantineEpifanov committed:
fix: fix api url

sergasent committed:
feat(booking): made single-select variant of Booking page

sergasent committed:
chore: resolve conflict

ConstantineEpifanov committed:
fix: fix urls

mintolime committed:
fix: fix popup window in react router

mintolime committed:
fix: fix popup window in react router

mintolime committed:
format :format  a code

ConstantineEpifanov committed:
fix: fix docker action

mintolime committed:
Merge branch 'develop' into auth-functionality

mintolime committed:
hotfix: fix bug popup close

ConstantineEpifanov committed:
feat: add server deploy

ConstantineEpifanov committed:
feat: add NotFoundError component

mintolime committed:
Merge branch 'auth-functionality' into develop

ConstantineEpifanov committed:
fix: fix Api token and Review error

vladdevjs committed:
feat: booking / profileData tabs

vladdevjs committed:
Resolved merge conflict

vladdevjs committed:
feat: ProfileDataTab first iteration

vladdevjs committed:
Merge pull request #69 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

vladdevjs committed:
fix: some fixes after design review

ConstantineEpifanov committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

vladdevjs committed:
fix: fixed console errors

sergasent committed:
fix(sectiontitle): added letter-spacing for SectionTitle

ConstantineEpifanov committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

ConstantineEpifanov committed:
fix: fix PromoItem styles

mintolime commented on src/App.jsx:
```diff
@@ -1,25 +1,183 @@
+    </CurrentUserContext.Provider>
   );
 }
 
```
```suggestion

```suggestion

```
```

sergasent committed:
fix(booking): booking page layout

ConstantineEpifanov committed:
fix: fix styles on Points

ConstantineEpifanov committed:
fix: fix styles on More button and fix onClick on Main page

yacax committed:
Create a function in api for coworking search and start to do implementation

ConstantineEpifanov committed:
fix: fix faq styles

ConstantineEpifanov committed:
fix: fix docker action and style in Rule

mintolime committed:
dev: created func for a errors api

mintolime committed:
fix: fixed error for email

yacax committed:
Add search function to search button

yacax committed:
Fix rerenders bug

mintolime committed:
refactor: refactored the hook openPopup

ConstantineEpifanov committed:
fix: fix styles on NotFoundError

yacax committed:
apply search

yacax committed:
Merge branch 'feature/coworking-search-and-filters' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters
apply search and filters

mintolime committed:
Merge branch 'feature/errors-func' into refactor-code

vladdevjs committed:
feat: booking send review and profileData redesign

yacax committed:
To transfer the server response to the list of coworking

yacax committed:
Hide the clear button

yacax committed:
Fix null error in location state

mintolime committed:
fix: fixed validation for password & email

mintolime committed:
Merge branch 'refactor-code' into develop

yacax committed:
Provide coworking by default with an empty input

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters

ConstantineEpifanov committed:
fix: fix More button

yacax committed:
Merge branch 'feature/coworking-search-and-filters' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters

sergasent committed:
feat(booking): made differ ranges logic, made price control

sergasent committed:
fix(booking): conflict in Constants.jsx

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

vladdevjs committed:
fix: switched to hook

vladdevjs committed:
Merge pull request #71 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

vladdevjs committed:
fix: some fixes in profile tabs

yacax committed:
Add empty string to the SortData

ConstantineEpifanov committed:
fix: fix styles

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

ConstantineEpifanov committed:
fix: fix styles (points, logo, favicon)

mintolime committed:
feat: realized the reset pass form

vladdevjs committed:
fix: many changes in profile

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

vladdevjs committed:
Merge pull request #72 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

sergasent committed:
feat(booking): added current time limitation for the time ranges

ConstantineEpifanov committed:
feat: add share button, fix urls

sergasent committed:
feat(booking): change select item logic

sergasent committed:
Merge branch 'feature/booking' into develop

ConstantineEpifanov committed:
fix: fix share, catch functions on Main and CoworkingList, fix 0 length search

ConstantineEpifanov committed:
fix: fix styles and booking on PointItem

mintolime committed:
refactor: refactored the peset pass form

mintolime committed:
Merge branch 'develop' into auth-functionality

sergasent committed:
feat: made order data transfer between PointsItem, Booking and Payments

sergasent committed:
Merge branch 'feature/booking-payment' into develop

vladdevjs committed:
fix: fixes in profiledataTab and design fixes

sergasent committed:
fix: changed Payments submit

mintolime committed:
fix: fix auth logic

mintolime committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

vladdevjs committed:
Merge pull request #73 from ConstantineEpifanov/feature/bookingTab

fix: fixes in profiledataTab and design fixes

ConstantineEpifanov committed:
fix: fix styles

FenixDeveloper commented on .storybook/preview.js:
```diff
@@ -1,4 +1,5 @@
 /** @type { import('@storybook/react').Preview } */
+
 const preview = {
```
Если что здесь можно добавлять глобальный контекст для историй, чтобы корректно работали компоненты. Нужно убедиться что окружения в сторибук и в приложении абсолютно идентичны — ключевой момент.

FenixDeveloper commented on README.md:
```diff
@@ -1,3 +1,39 @@
+  https://clck.ru/35vJy2
+
+- Автоматически обновляемая сборка Storybook на gh-pages:
+  https://constantineepifanov.github.io/book-it-coworking-frontend/
```
Страницы в сторибук не обязательно, разве что шаблонные. UI-kit можно было бы и расширить, вам же в помощь. Сейчас много ошибок, с ресурсом и контекстом исполнения.

Суть сторибука не чтобы усложнить вам жизнь, а чтобы упростить — но только при правильном подходе.
1. Идентичность окружения
2. Наиболее полно должны быть представлены самые наименьшие элементы UI-kit, то из чего строится все остальное.
3. Далее с повышением сложности компонент по необходимости, документируем в первую очередь наиболее часто переиспользуемые.

FenixDeveloper commented on package.json:
```diff
@@ -91,12 +97,15 @@
     "gh-pages": "^6.0.0",
     "husky": "^8.0.3",
     "lint-staged": "^14.0.1",
+    "node-sass": "^9.0.0",
```
Не нужен, достаточно только sass

FenixDeveloper commented on src/App.jsx:
```diff
@@ -1,25 +1,194 @@
-        </a>
-      </header>
-    </div>
+    <CurrentUserContext.Provider value={contextValue}>
```
Этот же контекст и в сторибуке нужен

FenixDeveloper commented on src/App.jsx:
```diff
@@ -1,25 +1,194 @@
+          isLoggedIn={isLoggedIn}
+          onLogout={handleLogout}
+        />
+        <Routes location={previousLocation || location}>
```
Я бы предложил вам отделить отсюда роуты, пусть приложение будет именно контейнером с окружением, которое получает роутер в пропсе. Так более гибко и вы сможете переиспользовать окружение например в сторибук.

FenixDeveloper commented on src/HOC/ProtectedRoute.js:
```diff
@@ -0,0 +1,16 @@
+  return isLoggedIn ? <Component {...props} /> : <Navigate to="/" replace />;
+};
+
+const AuthElement = ({ element: Component, ...props }) => {
```
И в чем смысл дублировать компонент? Суть у них одна и та же.

FenixDeveloper commented on src/components/BookingsCard/BookingsCard.jsx:
```diff
@@ -0,0 +1,232 @@
+      .finally(() => {});
+  };
+
+  const getPopupText = (booking) => {
```
Подобные функции лучше выносить за пределы компонента, аналогично константам.

FenixDeveloper commented on src/components/BookingsCard/BookingsCard.scss:
```diff
@@ -0,0 +1,151 @@
+  &__popup-text {
+    @include mp-null;
+    max-width: 641px;
+    font-size: 20px;
```
Большую часть типографики можно было бы вынести в миксины, сильно сократите код.

FenixDeveloper commented on src/components/ButtonsList/ButtonsList.jsx:
```diff
@@ -0,0 +1,229 @@
+  // };
+
+  // Получение класса для текущего элемента
+  const getClassName = (id) => {
```
посмотрите на либу clsx

FenixDeveloper commented on src/components/Calendar/Calendar.jsx:
```diff
@@ -0,0 +1,300 @@
+  maxDatesRange = CALENDAR_MAX_ALLOWED_DAYS,
+  extraRules = null,
+}) => {
+  const DAYS_NAMES = ["пн", "вт", "ср", "чт", "пт", "сб", "вс"];
```
в константы

FenixDeveloper commented on src/components/Calendar/Calendar.jsx:
```diff
@@ -0,0 +1,300 @@
+  return firstDay.getTime() > secondDay.getTime();
+};
+
+export const Calendar = ({
```
Вообще есть куча готовых календарей, можно было воспользоваться ими. Ну или хотя бы для вычислений взять что-то типа date-fns или Day.js, сформировать отдельно визуал и заполнить данными из них. Получилось много явно лишнего кода.

FenixDeveloper reviewed (approved):
По поводу сборки, почитайте еще про релизы github. Пихать сборку в image жирно, но можно собрать код и результат сохранить в релизе, а его потом забрать на сервер или также через действие закинуть на сервер.

В превью сторибука добавьте декоратор с контекстом (в идеале приложение, чтобы все стили и контексты были), это лучше чем отдельно на историях. Но в идеале обычно разделяют отображение и бизнес-логику и тогда это не требуется.

В миксины SCSS выделите как миниму типографику, еще неплохо было бы разные типы контейнеров — это закроет больше больше половины ваших стилей.

Начните использовать clsx для формирования классов, станет читабельнее код.

В целом неплохо, но нужно подрасчистить код:
1. Раскидать получше константы и утилиты
2. Уменьшить кодовую базу, повсеместно, особенно в стилях
3. Снизить связность по мере возможности
4. Разделить уровни отображения и бизнес-логики
5. Привести в порядок документацию

yacax committed:
Big bugs fix in search and filters

yacax committed:
Resolved merge conflicts from develop

ConstantineEpifanov commented:
> * Разделить уровни отображения и бизнес-логики
> * Привести в порядок документацию

Огромное спасибо за ревью @FenixDeveloper
Можно пояснить еще два момента. Про разделение уровней отображения, можно где-то почитать? И что имелось в виду под документацией, пользовательская к проекту, или комментарии по ходу кода?

mintolime committed:
feat: added a popup message info in auth

mintolime committed:
style: redacted styles for contact page

sergasent committed:
fix(booking): changed initial handle of the location state

mintolime committed:
style: changes header & contacts page

mintolime committed:
Merge branch 'refactor-code' into develop

vladdevjs committed:
feat: change password from profile

vladdevjs committed:
Merge pull request #74 from ConstantineEpifanov/feature/bookingTab

feat: change password from profile

mintolime committed:
fix:fixed info popup z-index

ConstantineEpifanov committed:
fix: fix styles on Coworking page

yacax committed:
fix(seachform & promo): fix search parameters

yacax committed:
Merge branch 'feature/coworking-search-and-filters' into develop

vladdevjs committed:
fix: design changes in profile

vladdevjs committed:
Merge pull request #75 from ConstantineEpifanov/feature/bookingTab

fix: design changes in profile

vladdevjs committed:
fix: design and func fixes

vladdevjs committed:
Merge pull request #76 from ConstantineEpifanov/feature/bookingTab

fix: design and func fixes

vladdevjs committed:
fix: updated bookings cancelling logics

vladdevjs committed:
Merge pull request #77 from ConstantineEpifanov/feature/bookingTab

fix: updated bookings cancelling logics

vladdevjs committed:
test: test deploy

vladdevjs committed:
Merge pull request #78 from ConstantineEpifanov/feature/bookingTab

test: test deploy

yacax committed:
feat(3): add MultiChoiceDropdown component, enhance usePagination hook,fix redundant submit issues

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into develop

sergasent committed:
fix(booking): fix Arseniy Booking loadning bug

sergasent committed:
fix(payment): make order bug, navigate back

sergasent committed:
fix(booking): working days extra rules for Calendar bug fixed

sergasent committed:
fix(payments): reload Payment state lose bug fixed

vladdevjs committed:
fix: design fixes

vladdevjs committed:
Merge pull request #79 from ConstantineEpifanov/feature/bookingTab

fix: design fixes

sergasent committed:
fix(payments): changed requests for Payments

sergasent committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

sergasent committed:
fix(booking): ordered visibility bug fixed

sergasent committed:
refactor: clear code

ConstantineEpifanov committed:
fix: fix styles

vladdevjs committed:
fix: a lot of fixes

vladdevjs committed:
Merge pull request #80 from ConstantineEpifanov/feature/bookingTab

fix: a lot of fixes

ConstantineEpifanov committed:
fix: qA fix

ConstantineEpifanov committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

yacax committed:
fix(3): fix coworking list loading. Added reset pagination and screen size handler

Fix the order in which the coworking list is loaded. Add pagination reset on new search. Add
automatic change of pagination parameters when changing the screen size.

mintolime committed:
feat: created a burger-menu

vladdevjs committed:
fixed conflict

vladdevjs committed:
Merge pull request #82 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

mintolime committed:
fix: fixed a popup window

mintolime committed:
Merge branch 'feature/burger-menu' into refactor-code

mintolime committed:
Merge branch 'develop' into refactor-code

mintolime committed:
refacror :formated code

mintolime committed:
Merge branch 'refactor-code' into develop

sergasent committed:
feat(payments): changed Payments popup layout

sergasent committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

vladdevjs committed:
fix: more responsive elements

vladdevjs committed:
Merge pull request #83 from ConstantineEpifanov/feature/bookingTab

fix: more responsive elements

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

yacax committed:
fix(2): minor fixes of UX elements (buttons)

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters

vladdevjs committed:
feat: avatarUpload component

vladdevjs committed:
Merge pull request #84 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

vladdevjs committed:
fix: local changes

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

yacax committed:
fix: merge branches

yacax committed:
Merge branch 'feature/coworking-search-and-filters' into develop

yacax committed:
fix(1): fix a bug in Button.scss

yacax committed:
Merge branch 'develop' into feature/coworking-search-and-filters

yacax committed:
fix(1): fix Button width

vladdevjs committed:
button small fix

vladdevjs committed:
fix brealpoint

vladdevjs committed:
Merge pull request #85 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

vladdevjs committed:
fix: breakpoint fix

yacax committed:
Add new button type.

yacax committed:
fix: save changes before merge

yacax committed:
Merge branch 'develop' into feature/coworking-search-and-filters

mintolime committed:
style: added adaptive for contact page

mintolime committed:
style: added footer mob

ConstantineEpifanov committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

ConstantineEpifanov committed:
fix: fix share url

vladdevjs committed:
feat: Avatar uploading

vladdevjs committed:
Merge pull request #86 from ConstantineEpifanov/feature/bookingTab

feat: Avatar uploading

vladdevjs committed:
fix: avatar hotfix

vladdevjs committed:
Merge pull request #88 from ConstantineEpifanov/feature/bookingTab

fix: avatar hotfix

ConstantineEpifanov committed:
fix: fix styles and typos

ConstantineEpifanov committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

yacax committed:
Make a mobile version of the co working List.

yacax committed:
feat: make a mobile version of coworking List

Make a mobile version of the co working List (in the Header, SearchForm part)

ConstantineEpifanov committed:
fix: fix reviews more button and styles

mintolime committed:
dev: added func close navigation

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters

yacax committed:
fix: merge branch with develope

mintolime committed:
Merge branch 'adaptive/footer_and_contacts' into feature/burger-menu

mintolime committed:
Merge branch 'feature/burger-menu' into develop

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into develop

yacax committed:
Merge branch 'feature/coworking-search-and-filters' into develop

yacax committed:
feat(1): add the filters button. Make mobile adaptation

vladdevjs committed:
feat: delete account

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

vladdevjs committed:
feat: pay orders from profile

vladdevjs committed:
Merge pull request #89 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

ConstantineEpifanov committed:
feat: add mobile styles for Main and FAQ pages

vladdevjs committed:
fix: a bulk of fixes

vladdevjs committed:
Merge pull request #90 from ConstantineEpifanov/feature/bookingTab

fix: a bulk of fixes

yacax committed:
Merge branch 'develop' of github.com-yacax:ConstantineEpifanov/book-it-coworking-frontend into feature/coworking-search-and-filters

ConstantineEpifanov committed:
chore: change equipment title

sergasent committed:
feat: changed Booking time range selection, made Calendar adaptive

sergasent committed:
Merge branch 'feature/calendar-adaptive' into develop

yacax committed:
feat(2): make a mobile version of the main page. Make a carousel of coworkings

yacax committed:
Merge branch 'develop' into feature/coworking-search-and-filters

yacax committed:
fix(2): set up a carousel of coworking spaces. Fix the "More" button

mintolime committed:
Merge branch 'code-refactor' into develop

mintolime committed:
style: adapted 404 page

sergasent committed:
feat: made Booking mostly adaptive

sergasent committed:
Merge branch 'feature/booking-adaptive' into develop

vladdevjs committed:
fix: fine-tuning Favorites and more

vladdevjs committed:
Merge pull request #92 from ConstantineEpifanov/feature/bookingTab

fix: fine-tuning Favorites and more

vladdevjs committed:
fixed profile and security tabs

vladdevjs committed:
Merge branch 'develop' into feature/bookingTab

vladdevjs committed:
fix: small fixes bookingCard

vladdevjs committed:
Merge pull request #93 from ConstantineEpifanov/feature/bookingTab

Feature/booking tab

yacax committed:
fix(1): replace the coworking carousel with a list of cards on the desktop

yacax committed:
Merge branch 'develop' into feature/coworking-search-and-filters

vladdevjs committed:
refactor: refactor Contacts page, optimized images

vladdevjs committed:
Merge branch 'feature/bookingTab' into develop

sergasent committed:
fix: bug with long time-lists in Booking fixed

sergasent committed:
Merge branch 'develop' of https://github.com/ConstantineEpifanov/book-it-coworking-frontend into develop

sergasent committed:
refactor: changed place for setting buttons-list vars in Booking

vladdevjs committed:
feat: added author images for reviews

vladdevjs committed:
Merge pull request #95 from ConstantineEpifanov/feature/bookingTab

feat: added author images for reviews

ConstantineEpifanov committed:
fix: qA fixes

ConstantineEpifanov committed:
fix: fix second get locations

sergasent committed:
feat: added mobile popup for choose date and time in Booking

sergasent committed:
Merge pull request #96 from ConstantineEpifanov/feature/date-popup

feat: added mobile popup for choose date and time in Booking

sergasent committed:
fix: booking mobile popup layout bugs fixed

mintolime committed:
feat: added 500 page

mintolime committed:
fix: fixed a 500 page
